hilti/runtime/include/types/stream.h

```diff
@@ -156,13 +185,30 @@ class Chunk {
         return i;
     }
 
+    // Creates a new copy of the data internally if the chunk is currently not
+    // owning it. On return, is guaranteed to now own the data.
+    void makeOwning() {
+        if ( _size == 0 || _allocated > 0 || ! _data )
```

Comment:
I'm confused where that line `191` is coming from, it's currently looking like this:

```
    void makeOwning() {
        if ( _allocated > 0 || ! _data )
            return;
    ....
```

And none of the individual commits seems to introduce the line like above either.

The current check isn't the same as `isGap()`: the latter is using the explicit gap flag. (Note that `_data == null` by itself isn't marking a gap; that's so that we can have a regular zero-sized chunk where the caller may just pass us a nullptr to "own" (or even `new` with size 0 could return null, I guess)).

Reply:
Ah, I see it now, I messed something up in my local versions. Answer remains the same, though. :-)

Reply:
Actually it doesn't: I realize now what happened. I have a local version where I had tried one more thing but discarded that approach; but then didn't reset my local branch, so was looking at slightly different code. There's no explicit gap flag, but the condition still needs to remain as is.